src/hashedSecret.ts

```diff
@@ -10,41 +10,102 @@ import { publicKeyConvert } from 'secp256k1'
 
 export const GIANT_STEP_WIDTH = 10000
 export const TOTAL_ITERATIONS = 100000
-
 export const HASHED_SECRET_WIDTH = 27
+export const EMPTY_HASHED_SECRET = new Uint8Array(HASHED_SECRET_WIDTH).fill(0x00)
 
 class HashedSecret {
-  public _onChainValuesInitialized: boolean
+  constructor(private coreConnector: HoprEthereum) {}
+
+  /**
+   * @returns a promise that resolves to a Hash if secret is found
+   */
+  private async getOnChainSecret(): Promise<Hash | undefined> {
+    const address = (await this.coreConnector.account.address).toHex()
+
+    return this.coreConnector.hoprChannels.methods
+      .accounts(address)
+      .call()
+      .then((res) => {
+        const hashedSecret = stringToU8a(res.hashedSecret)
+
+        // true if this string is an empty bytes32
+        if (u8aEquals(hashedSecret, EMPTY_HASHED_SECRET)) {
+          return undefined
```

Comment:
My problem with this is that we are throwing away any runtime capabilities TypeScript is giving us. We are pretty much saying "mah, don't know what's this". Don't we have a better Type we can use?